crates/test-utils/src/lib.rs

```diff
@@ -1,46 +1,40 @@
-use std::sync::atomic::{AtomicUsize, Ordering};
-
 use arbitrary::{Arbitrary, Unstructured};
-use rand::{rngs::OsRng, RngCore};
+use rand::{thread_rng, RngCore};
 
 pub mod bitcoin;
 pub mod bridge;
 pub mod evm_ee;
 pub mod l2;
 
-const ARB_GEN_LEN: usize = 1 << 24; // 16 MiB
+// Smaller buffer size as compared to 2^24
+const ARB_GEN_LEN: usize = 128;
 
 pub struct ArbitraryGenerator {
-    buf: Vec<u8>,
-    off: AtomicUsize,
+    rng: rand::rngs::ThreadRng, // Thread-local RNG
```

Comment:
What's the rationale for the use of `ThreadRng` versus `OsRng`? We've been standardizing on the use of `OsRng` elsewhere.

Reply:
^ pinging @delbonis as well, given recent approval

Reply:
It's faster, right?  I don't think it really matters since it's only being used in tests.

Reply:
In theory it's faster, but I think it's worth being consistent to avoid it being copy-pasted outside of this context.

Reply:
Something something premature optimization...

Reply:
> being consistent to avoid it being copy-pasted outside of this context.

Sorry, I could not get the context of it. @AaronFeickert

Reply:
Yes we can move it outside of the `struct`. Let it take a (mutable) reference to an PRNG, and the user bring either a fast one (in the case of testing and fuzzing) or a slow and safe (CSPRNG) one.

Reply:
Just put a trait bound on https://docs.rs/rand/latest/rand/trait.RngCore.html and let the user bring either a https://docs.rs/rand/latest/rand/rngs/struct.OsRng.html which impls `RngCore` or something faster like https://docs.rs/rand/latest/rand/trait.RngCore.html#impl-RngCore-for-ChaCha12Rng that also impls `RngCore`.

Reply:
I recommend requiring `CryptoRngCore` (which is just `RngCore + CryptoRng`) to mitigate against the unintentional misuse of obviously bad generators. Even for fast testing operations, surely something like `ChaCha8Rng` would be performant enough.

Also note that `CryptoRngCore` does _not_ guarantee the generator's output is of high entropy!

Reply:
I mean, it's only used for tests, there's no such thing as a bad generator.

Reply:
[orly](https://xkcd.com/221/)

Reply:
I kid, but I think this highlights a suboptimal design for which better design patterns exist.

Reply:
Plus I am generally a fan of having tests reflect optimal behavior as much as possible. You never know who will take a design pattern from a test and use it where they shouldn't.